### PR TITLE
Partial transformers documentation

### PR DESCRIPTION
**Pull request by krzemin**

### CODE REVIEW

File: chimney/src/main/scala/io/scalaland/chimney/PartialTransformer.scala [hunk 1]
```diff
@@ -13,7 +13,7 @@ import scala.language.experimental.macros
   * @tparam From type of input value
   * @tparam To   type of output value
   */
-trait PartialTransformer[From, To] { self =>
+trait PartialTransformer[From, To] {
 
   def transform(src: From, failFast: Boolean): partial.Result[To]
```
Comment (MateuszKubuszok) [inline chimney/src/main/scala/io/scalaland/chimney/PartialTransformer.scala:18]: Might be good to describe these parameters, if for anything else to document how user should treat `failFast` parameter if they decide to implement this trait without any helpers. Probably with a mention that is is encouraged to use a dedicated factory method to not go through all the hassle.

File: docs/source/partial-transformers/cats-integration.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+--------
+
+Cats integration module contains the following stuff:
+
+* type classes instances for partial transformers data structures
+
+    * ``Applicative`` instance for ``partial.Result``
+    * ``Semigroup`` instance for ``partial.Result.Errors``
+
+* integration with ``Validated`` data type
```
Comment (MateuszKubuszok) [inline docs/source/partial-transformers/cats-integration.rst:38]: I'd add in parenthesis: (and `ValidatedNec` and `ValidatedNel`)

File: docs/source/partial-transformers/cats-integration.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+Example
+-------
+
+Let's have a look at how to integrate :ref:`partial-transformers` with cats ``Validated``.
```
Comment (MateuszKubuszok) [inline docs/source/partial-transformers/cats-integration.rst:43]: Cats'

File: docs/source/partial-transformers/cats-integration.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
+.. _partial-cats-integration:
+
+Cats integration
+================
+
+Chimney provides `Cats <https://typelevel.org/cats>`_ library integration module
+for partial transformers.
+To include it to your SBT project, add the following line to your ``build.sbt``:
```
Comment (MateuszKubuszok) [inline docs/source/partial-transformers/cats-integration.rst:8]: Formally sbt is capitalized as "sbt" (all lower case), so we might want to use that in docs.

File: docs/source/partial-transformers/migrating-from-lifted.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  In case you are using lifted transformers, we strongly recommend migration to :ref:`partial-transformers`.
+
+
+Design differences
+------------------
+
+Let's have a look at type signatures of both lifted and partial transformers.
+
+.. code-block:: scala
+
```
Comment (MateuszKubuszok) [inline docs/source/partial-transformers/migrating-from-lifted.rst:21]: `package io.scalaland.chimney`?

File: docs/source/partial-transformers/partial-transformers.rst [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+    val resultOk = okForm
+      .intoPartial[RegisteredUser]
+      .withFieldComputed(_.passwordHash, form => hashpw(form.password))
+      .withFieldComputedPartial(_.age, form => partial.Result.fromOption(form.age.toIntOption))
+      .transform
+    // resultOk: partial.Result[RegisteredUser] = Value(
+    //   RegisteredUser("john@example.com", "John", "...", 40)
+    // )
+
+Partial transformation computes into special result type ``partial.Result``. There are few methods that are
```
Comment (MateuszKubuszok) [inline docs/source/partial-transformers/partial-transformers.rst:43]: "returns a special result type"?

File: docs/source/partial-transformers/migrating-from-lifted.rst [hunk 7]
```diff
@@ -0,0 +1,98 @@
+.. _migrating-from-lifted:
+
+Migrating from Lifted Transformers
+==================================
+
+Chimney's :ref:`lifted-transformers` were historically first experimental attempt
```
Comment (MateuszKubuszok) [inline docs/source/partial-transformers/migrating-from-lifted.rst:6]: the first

File: docs/source/partial-transformers/migrating-from-lifted.rst [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+to express transformations that may potentially fail. Despite their great expressiveness, they were
+lacking several basic features and had few design flaws that didn't make them attractive for wider usage.
```
Comment (MateuszKubuszok) [inline docs/source/partial-transformers/migrating-from-lifted.rst:8]: and had a few design flaws that  make them unattractive/difficult for wider audience.

File: docs/source/partial-transformers/migrating-from-lifted.rst [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+    * ``result.asOption``
+
+    * ``result.asEither``
+
+    * ``result.asErrorPathMessages``
+
+    * other data structures using :ref:`partial-cats-integration`
+
+.. TODO: provide github links to the paragraph below
```
Comment (MateuszKubuszok) [inline docs/source/partial-transformers/migrating-from-lifted.rst:95]: I believe this link will be https://github.com/scalalandio/chimney/tree/master/chimney/src/test/scala/io/scalaland/chimney - at least until we remove the lifted transformers but then this section will be removed from the docs as well.

(If we decide to keep docs versioned we might decide to use something like https://github.com/scalalandio/chimney/tree/v0.7.0/chimney/src/test/scala/io/scalaland/chimney to make the link permalink).

File: docs/source/transformers/unsafe-options.rst [hunk 10]
```diff
@@ -1,5 +1,5 @@
-Unsafe options transformation
-=============================
+Unsafe options ``(deprecated)``
+===============================
 
 .. warning::
```
Comment (MateuszKubuszok) [inline docs/source/transformers/unsafe-options.rst:4]: We might want to add a mention that this flag is scheduled for remove in 0.8.0 and migration to partial transformers is recommended (and users that want to watch their world burn can implement `[A] => Transformer[Option[A], A]`).

File: docs/source/lifted-transformers/lifted-transformers.rst [hunk 11]
```diff
@@ -1,7 +1,7 @@
 .. _lifted-transformers:
 
-Lifted transformers
-===================
+Lifted Transformers ``(deprecated)``
+====================================
 
 .. warning::
```
Comment (MateuszKubuszok) [inline docs/source/lifted-transformers/lifted-transformers.rst:6]: Same as unsafeOption - we might explain in the warning that this is scheduled for deletion in probably 0.8.0

File: docs/source/partial-transformers/partial-transformers.rst [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+
+Custom error messages
+---------------------
+
+So far we were receiving `EmptyValue` error, as we just provided ``Option[Int]`` value to
+``withFieldComputedPartial``, which can't represent any more detailed error information.
+
+.. code-block:: scala
+
+    import scala.util.{Try, Success, Failure}
```
Comment (MateuszKubuszok) [inline docs/source/partial-transformers/partial-transformers.rst:100]: `import io.scalaland.chimney.dsl._`

**Review by MateuszKubuszok — CHANGES_REQUESTED** [review]

File: chimney/src/main/scala/io/scalaland/chimney/PartialTransformer.scala (hunk 1 above)
Comment (krzemin) [reply chimney/src/main/scala/io/scalaland/chimney/PartialTransformer.scala:18]: will do, when working on scaladoc

File: docs/source/partial-transformers/migrating-from-lifted.rst (hunk 5 above)
Comment (krzemin) [reply docs/source/partial-transformers/migrating-from-lifted.rst:21]: why not

File: docs/source/transformers/unsafe-options.rst (hunk 10 above)
Comment (krzemin) [reply docs/source/transformers/unsafe-options.rst:4]: I don't want to make any promises when this is going to be implemented.

And below there is already:

> Consider using :ref:`partial-transformers` instead.

File: docs/source/lifted-transformers/lifted-transformers.rst (hunk 11 above)
Comment (krzemin) [reply docs/source/lifted-transformers/lifted-transformers.rst:6]: Same as above, no promises as general rule.

**Review by MateuszKubuszok — APPROVED** [review]

**Review by MateuszKubuszok — APPROVED** [review]

**Pull request merged**